File: leetcode/golang/problems/559. N 叉树的最大深度.go
package problems

func maxDepth(root *Node) (ans int) {
    if root == nil {
        return
    }
	q := []*Node{}
	q = append(q, root)
	for len(q) > 0 {
        size := len(q)
        ans++
        for size > 0 {
            node := q[0]
            q = q[1:]
            for i := 0; i < len(node.Children); i++ {
                q = append(q, node.Children[i])
            }
            size--
        }
	}
	return
}

File: leetcode/javascript/559. N 叉树的最大深度.js
// Definition for a Node.
class Node {
    constructor(val, children) {
        this.val = val;
        this.children = children;
    }
};

/**
 * @param {Node|null} root
 * @return {number}
 */
 var maxDepth = function(root) {
    if (!root) {
        return 0;
    }
    let depth = 0;
    const children = root.children;
    for (const child of children) {
        const childepth = maxDepth(child);
        depth = Math.max(depth, childepth);
    }
    return depth + 1;
};